pkg/plugin/system/aspectplugins.go

```diff
@@ -114,5 +117,16 @@ func (p *parser) Parse(aspectpluginsPath string) ([]AspectPlugin, error) {
 	if err := p.yamlUnmarshalStrict(aspectpluginsData, &aspectplugins); err != nil {
 		return nil, fmt.Errorf("failed to parse .aspectplugins: %w", err)
 	}
-	return aspectplugins, nil
+
+	var processedPlugins []AspectPlugin
+	for _, plugin := range aspectplugins {
+		marshalledProperties, err := p.yamlMarshal(plugin.PropertiesMap)
+		if err != nil {
+			return nil, fmt.Errorf("failed to parse .aspectplugins: %w", err)
+		}
+		plugin.properties = marshalledProperties
```

Comment:
Continuing the suggestion above, see how it looks clearer?
```suggestion
		propertiesBytes, err := p.yamlMarshal(plugin.Properties)
		if err != nil {
			return nil, fmt.Errorf("failed to parse .aspectplugins: %w", err)
		}
		plugin.propertiesBytes = propertiesBytes
```

Reply:
Yes it does look cleaner like this. I agree